packages/components/src/utils/strings.ts

```diff
@@ -81,3 +82,29 @@ export const normalizeTextString = ( value: string ): string => {
 export function escapeRegExp( string: string ): string {
 	return string.replace( /[\\^$.*+?()[\]{}|]/g, '\\$&' );
 }
+
+/**
+ * Converts any string to kebab case.
+ *
+ * @param {string} str String to convert.
+ *
+ * @return {string} Kebab-cased string
+ */
+export function kebabCase( str: any ) {
```

Comment:
I created a new function to take the numbers into account and kebab case them. This is exactly [the same function](https://github.com/WordPress/gutenberg/blob/f987124f44e5d22d64a8e232759f5a213c329bc8/packages/block-editor/src/utils/object.js#L17) found in the block editor package.

Reply:
We should not be duplicating code like that throughout the codebase. 

One way we could approach it is by moving some code around. `@wordpress/components` is a dependency of `@wordpress/block-editor`, so perhaps we can move it to `@wordpress/components`, maybe as part of the privately exposed APIs? Then the block editor package can re-export it, or we can change all exports to consume directly from the `@wordpress/components` package. WDYT?

Reply:
> so perhaps we can move it to `@wordpress/components`, maybe as part of the privately exposed APIs?

Indee, in that case there is a lot of code to change, so it might be better to address it as a separate PR and then come back to this PR.

Moving this code means that the `kebabKase()` function will disappear from the `@wordpress/block-editor` package, but since it is a private API, does that mean we don't have to think about backward compatibility?

Reply:
Yes 👍 it is still a compromise, but I'm happy with it since it operates solely with private APIs.

Thanks for considering it!